app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Super/Config/Simple.php

```diff
@@ -108,7 +110,9 @@ protected function _prepareForm()
         }
 
         /* Configurable attributes */
-        $usedAttributes = $this->_getProduct()->getTypeInstance(true)->getUsedProductAttributes($this->_getProduct());
+        /** @var Mage_Catalog_Model_Product_Type_Configurable $productType */
+        $productType = $this->_getProduct()->getTypeInstance(true);
+        $usedAttributes = $productType->getUsedProductAttributes($this->_getProduct());
         foreach ($usedAttributes as $attribute) {
```

Comment:
```suggestion
        /** @var Mage_Eav_Model_Attribute $attribute */
        foreach ($usedAttributes as $attribute) {
```

Reply:
Why this? `Mage_Catalog_Model_Product_Type_Configurable::getUsedProductAttributes()` returns ...

```
     * @return Mage_Catalog_Model_Product_Type_Configurable_Attribute[]
```

![pr-2605](https://user-images.githubusercontent.com/5022236/191138422-7e86c369-0a84-43bb-afa6-69a2e48dabb3.PNG)

Reply:
My bad. You are right, it should be `Mage_Catalog_Model_Product_Type_Configurable_Attribute`.

Reply:
Wait, I think `$attribute` is  `Mage_Catalog_Model_Resource_Eav_Attribute`:

Reply:
Yep. Is it possibie to update that suggestion? Return turn should be changed here:

```
     * @param  Mage_Catalog_Model_Product $product
     * @return Mage_Catalog_Model_Resource_Eav_Attribute[]
     */
    public function getUsedProductAttributes($product = null)
```